METIS/default.yaml

```diff
@@ -24,6 +24,9 @@ properties:
   pupil_angle: 0
   dit: 1
   ndit: 1
+  auto_exposure:
+    exptime: None     # seconds, overrides !OBS.dit and !OBS.ndit
+    fill_frac: 0.75   # fraction to which full well is filled in one DIT
```

Comment:
Theoretically this could be >1 right? Then it would be possible to also use auto exposure for frames where it is acceptable for the brightest source to be saturated, or?

Reply:
That's right, the effect does test for `fill_frac` to be less than 1. I think we can rely on the user to know what they're doing. A test for non-negativity might be justified, though.

Reply:
I think we should try to make ScopeSim assist the user as much as possible in "being sensible" here: 

1. test whether the peak intensity [e-/s/pix] times the DIT [s] is more than the full well x the allowable fill fraction. 
2. If this is the case, all is good
3. If not, then:
    - if the linearity option is on, continue but put out a prominent warning message that the source saturates (goes into the non-linear regime) so that photometry will be unreliable (perhaps indicate which pixels are affected?)
    - if the linearity option is off, refuse to continue, put out an error message saying that the source saturates, and that in order to get output at all, the linearity option needs to be switched on (say how, and stress that this will still give unreliable photometry). But really, that the user should consider changing the setup (e.g. use an ND filter, or use a coronagraphic observation; cf. discussions with Liege coupling ScopeSim with HEEPS).

Reply:
1, 2, and 3a are already implemented in Scopesim. The code tests whether the automatically determined DIT is less than MINDIT. If it is, then DIT is set to MINDIT and the warning is issued. Indicating the affected pixels would be require non-negligible coding work and is something that seems quite alien to Scopesim. As for 3b, this seems rather intrusive - it is not obvious how the user would get around that blockage should they wish to do so. It seems preferable to leave the (non-)linearity on by default. It is then an effort for the user to switch it off, and we assume they only do it for a reason.

Reply:
Thanks! Switching the linearity on by default, with a warning if a simulation goes into the non-linear regime, and requiring special action if an expert user wants to circumvent that for test purposes, makes sense and is certainly more elegant. Perhaps we should always include the "range of where we are on the linearity curve" with any simulation (like: lowest flux level we are at 99.x1 percent linearity, highest flux level we are at 98.x2 percent). Perhaps have a text file with "details" that is created then running a simulation (or packed into the fits header); we could put information in there that in most cases the user won't care too much about, but for users getting into the nitty-gritty or trying to do high accuracy applications may/should care about). Something for future consideration perhaps.